File: Backen_Autobus/functions/src/models/flotas_model.ts
/* Modelo representa el concepto de flota */
/*============ Flota ============*/
export interface Flota {
    idFlota? :string,
    nomCoperativa :string,
    descripcion :string,
    imagen: string,
    
}

export function Flota(data :any, id?:string){
    const { nomCoperativa, descripcion, imagen } = data;

    let object :Flota = { 
        idFlota: id,                
        nomCoperativa :nomCoperativa,
        descripcion: descripcion,
        imagen: imagen
    };
    return object;
}

File: App_Admin/lib/src/models/flotas_model.dart
// To parse this JSON data, do
//
//     final Flotas = FlotasFromJson(jsonString);

import 'dart:convert';

Flota flotasFromJson(String str) => Flota.fromJson(json.decode(str));

String flotasToJson(Flota data) => json.encode(data.toJson());

class Flota {
  Flota({
    this.idFlota,
    this.nomCoperativa,
    this.descripcion,
    this.imagen,
  });

  String? idFlota;
  String? nomCoperativa;
  String? descripcion;
  String? imagen;

  factory Flota.fromJson(Map<String, dynamic> json) => Flota(
        idFlota: json["idFlota"],
        nomCoperativa: json["nomCoperativa"],
        descripcion: json["descripcion"],
        imagen: json["imagen"],
      );

  factory Flota.created() => Flota();

  Map<String, dynamic> toJson() => {
        "idFlota": idFlota,
        "nomCoperativa": nomCoperativa,
        "descripcion": descripcion,
        "imagen": imagen
      };
}
